### DISCUSSION TITLE
MLE prefix mismatch between FTD and SED

### DISCUSSION THREAD

**Discussion by ian2739** [openthread/openthread#7876] [Q&A] [answered]
I have a FTD and two SEDs running and recently after programming and reconfiguring their thread settings (panid, networkkey, channel) one of the SEDs stopped responding to coap requests from the FTD. The MLE prefix of this SED doesn't match the MLE prefix of the FTD so the requests are sent to the wrong address. This SED is still visible to the FTD as it appears in the child table and is regularly issuing data requests to the FTD.
Their IPs are below.

**end device not responding**
fdc8:8e00:201c:0:0:ff:fe00:3402
fdc8:8e00:201c:0:f8c7:55b2:8f:cfea
fe80:0:0:0:f4b8:a013:b908:d582

**end device responding**
fdde:ad00:beef:0:0:ff:fe00:3403
fdde:ad00:beef:0:ee36:b4e3:18c4:4ad7
fe80:0:0:0:44a:42de:df55:3d5

**router ips**
fdde:ad00:beef:0:0:ff:fe00:fc00
fdde:ad00:beef:0:0:ff:fe00:3400
fdde:ad00:beef:0:3718:4248:7b40:4884
fe80:0:0:0:a483:284a:75ff:8422

Any thoughts on how the end device could obtain the incorrect MLE prefix?
We do have other openthread networks operating. Could this SED have joined another network, obtained that mle prefix and not updated it once it connected to my network?

- Zephyr revision 2.7.99-ncs1
- Openthread revision b7aa7686d3f4e326bd25abc234b209159977c26d
- nrf52840 hardware

**Comment by jwhui** [comment, marked as answer]
The mesh-local prefix configuration is part of the Operational Dataset, which includes things like network key, channel, etc.

A device joining a new Thread network can obtain the full Operational Dataset from a neighboring device, but the Active Timestamp needs to be larger than what the device has locally.

I suggest taking a look at the [Dataset Quick Start guide](https://github.com/openthread/openthread/blob/main/src/cli/README_DATASET.md#quick-start).

**Reply by ian2739** [reply]
Thanks for the help. It looks like the issue was related to the Active Timestamp and that we were temporarily using up time instead of actual timestamps.
When the FTD has the latest dataset, I am seeing it pushed to the MTDs along with the mesh-local prefix and everything works. Alternatively, when the MTD has the latest dataset, it's pushed to the FTD except for the mesh-local prefix so they are unable to communicate. Is this behavior intended? Should the FTD always contain the latest dataset to avoid this?

**Reply by jwhui** [reply]
If the MTD has a higher Active Timestamp value, the FTD(s) should eventually adopt the new operational dataset, which includes the mesh-local prefix. Can you provide steps to reproduce the issue you are seeing?
